### PR TITLE
Fix migrate schedule config procedure

### LINKED ISSUE #216: Error for reports with schedules after database migration to v1.0.0
## Describe the bug

The column config ist NULL in table schedule after applying migration script for MariaDB 10.5.22 on Oracle Linux 8.
This leads to errors in the WebUI and reporting daemon referencing library/Reporting/Schedule.php line 58.

## To Reproduce

1. Update to v 1.0.0
2. Execute Migration Script

## Expected behavior

The migration should convert the 

## Your Environment

* Module version: 1.0.0
* Dependent module versions: 
* Icinga Web 2 version and modules (System - About): 2.12
* Web browser and version: 
* Icinga 2 version used (`icinga2 --version`):
* PHP version used (`php --version`):
* Server operating system and version: Oracle Linux 8

## Additional context

The cause for the failing migration is the use of the named timezones for the function "CONVERT_TZ".
As the documentation states (https://mariadb.com/kb/en/convert_tz/), those are only available after loading them with "mysql_tzinfo_to_sql".
After restoring the previous version of the table and executing the migration script with
CONVERT_TZ(FROM_UNIXTIME(schedule_start / 1000), @@session.TIME_ZONE, 'UTC')
replaced by
CONVERT_TZ(FROM_UNIXTIME(schedule_start / 1000), '+02:00', '+00:00')
the migration worked.
As a sidenote, there is another small error. The DATE_FORMAT "%u" is for weeknumbers. You probably ment "%f" (https://mariadb.com/kb/en/date_format/).

### PR DESCRIPTION
**Pull request by yhabteab**
fixes #216

### CODE REVIEW

**Review by nilmerg — DISMISSED** [review]

**Comment by yhabteab** [comment]
Tested also UI based migrations.

File: doc/80-Upgrading.md [hunk 1]
```diff
@@ -8,8 +8,13 @@ Usually the only manual steps involved are schema updates for the database.
 Icinga Reporting version 1.0.0 requires a schema update for the database.
 If you're already using Icinga Web version `>= 2.12`, then you don't need to perform any of these steps manually.
 Icinga Web provides you the ability to perform such migrations in a simple way. You may be familiar with such an
-automation if you're an Icinga Director user. For those who are not using the latest version of Icinga Web, please
-follow the instructions below.
+automation if you're an Icinga Director user. For those who are not using the latest version of Icinga Web, please follow the instructions below.
```
Comment (nilmerg) [inline doc/80-Upgrading.md:11]: Please break this line as was previously the case

File: schema/mysql-upgrades/1.0.0.sql [hunk 2]
```diff
@@ -13,6 +15,13 @@ BEGIN
   DECLARE schedule CURSOR FOR SELECT id, start, frequency, config FROM schedule;
   DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = 1;
 
+  -- Determine the current session time zone name
+  SELECT IF(@@session.TIME_ZONE = 'SYSTEM', @@system_time_zone, @@session.TIME_ZONE) INTO session_time_zone;
+
+  IF session_time_zone NOT LIKE '+%:%' AND CONVERT_TZ(FROM_UNIXTIME(1699903042), session_time_zone, '+00:00') IS NULL THEN
```
Comment (nilmerg) [inline schema/mysql-upgrades/1.0.0.sql:21]: offsets may also be negative

File: doc/80-Upgrading.md [hunk 3]
```diff
@@ -11,6 +11,12 @@ Icinga Web provides you the ability to perform such migrations in a simple way.
 automation if you're an Icinga Director user. For those who are not using the latest version of Icinga Web, please
 follow the instructions below.
 
+> **Note**
+>
+> If you're not using Icinga Web migration automation, you may [populate](https://dev.mysql.com/doc/refman/8.0/en/time-zone-support.html#time-zone-installation)
```
Comment (nilmerg) [inline doc/80-Upgrading.md:16]: *you may need to*

File: doc/80-Upgrading.md [hunk 4]
```diff
@@ -11,6 +11,12 @@ Icinga Web provides you the ability to perform such migrations in a simple way.
...
+> all the system named time zone information into your MSQL/MariaDB server. Otherwise, the migration will not succeed
```
Comment (nilmerg) [inline doc/80-Upgrading.md:17]: *the migration may not*

File: doc/80-Upgrading.md [hunk 5]
```diff
@@ -11,6 +11,12 @@ Icinga Web provides you the ability to perform such migrations in a simple way.
...
+> and will produce an SQL error.
```
Comment (nilmerg) [inline doc/80-Upgrading.md:18]: you can remove this imho, that the migration might not succeed is sufficient.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: schema/mysql-upgrades/1.0.0.sql [hunk 6]
```diff
@@ -13,6 +15,13 @@ BEGIN
   DECLARE schedule CURSOR FOR SELECT id, start, frequency, config FROM schedule;
   DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = 1;
 
+  -- Determine the current session time zone name
+  SELECT IF(@@session.TIME_ZONE = 'SYSTEM', @@system_time_zone, @@session.TIME_ZONE) INTO session_time_zone;
+
+  IF (session_time_zone NOT LIKE '+%:%' OR session_time_zone NOT LIKE '-%:%') AND CONVERT_TZ(FROM_UNIXTIME(1699903042), session_time_zone, '+00:00') IS NULL THEN
```
Comment (nilmerg) [inline schema/mysql-upgrades/1.0.0.sql:21]: If it's *like* +%:%, it's *unlike* -%:%
If it's *like* -%:%, it's *unlike* +%:%

You want *if unlike +%:% and unlike -%:% and convert_tz(...) is null*, don't you?

**Review by nilmerg — APPROVED** [review]

**Pull request merged**